Mark some projects as non-packable or non-shippable

dougbu opened the pull request:
- used `$(IsUnitTestProject)` in src/Razor/test; slightly more correct

dougbu committed:
Mark some projects as non-packable or non-shippable
- used `$(IsUnitTestProject)` in src/Razor/test; slightly more correct

dougbu commented:
@Pilchie this changes what we ship. So I'm asking for your approval. Please&hellip;

wtgodbe commented on src/Razor/test/Directory.Build.props:
```diff
@@ -0,0 +1,8 @@
+
+  <PropertyGroup>
+    <!-- Projects in this folder don't follow the usual naming conventions for tests. -->
+    <IsUnitTestProject>false</IsUnitTestProject>
```
Why false? Aren't these files tests?

dougbu replied:
Because I made a mistake

dougbu replied:
```suggestion
    <IsUnitTestProject>true</IsUnitTestProject>
```

wtgodbe reviewed (approved):
Looks good pending the 1 question

dougbu commented on src/Components/WebAssembly/Sdk/src/Microsoft.NET.Sdk.BlazorWebAssembly.csproj:
```diff
@@ -12,6 +12,8 @@
     <!-- Need to build this project in source build -->
     <ExcludeFromSourceBuild>false</ExcludeFromSourceBuild>
+
+    <IsShipping>false</IsShipping>
```
```suggestion
    <IsShipping>true</IsShipping>
```

dougbu commented on src/Components/WebAssembly/Sdk/src/Microsoft.NET.Sdk.BlazorWebAssembly.csproj:
```diff
@@ -12,6 +12,8 @@
     <!-- Need to build this project in source build -->
     <ExcludeFromSourceBuild>false</ExcludeFromSourceBuild>
+
+    <IsShipping>true</IsShipping>
```
```suggestion
    <IsShipping>false</IsShipping>
```
Need to read more

Pilchie commented:
Seems good to me.  Approved for RC1 pending CI

dougbu committed:
Mark src/Razor/test/ projects as test assets instead
- don't seem to have any actual tests in this folder (unless I missed something)

dougbu commented:
/azp run
